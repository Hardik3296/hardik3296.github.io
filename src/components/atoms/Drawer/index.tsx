import { Menu, MenuItem } from "@material-ui/core";
import { GitHub, LinkedIn, Mail, Menu as MenuIcon } from "@material-ui/icons";
import React, { Dispatch, SetStateAction, useContext } from "react";
import { ThemeContext, ThemeInterface } from "../../../utils/contexts/ThemeContext";
import GeeksForGeeks from "../../../assets/svgs/GeeksForGeeksSvg";

const DrawerComponent = (): JSX.Element => {
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
   const [theme, _setTheme] = useContext<[ThemeInterface, Dispatch<SetStateAction<ThemeInterface>>]>(ThemeContext);
   const iconStyle = { ...theme.text, marginLeft: 20 };

   const handleClose = () => {
      setAnchorEl(null);
   }
   const handleClick = (event: any) => {
      setAnchorEl(anchorEl ? null : event.currentTarget);
   }
   const handleIconClick = (link: string, target: string): void => {
      handleClose();
      window.open(link, target);
   }
   return (
      <>
         <MenuIcon onClick={handleClick} style={{ color: theme.text.color }} />
         <Menu open={Boolean(anchorEl)} onClose={handleClose} anchorEl={anchorEl} MenuListProps={{ disablePadding: true, autoFocus: false }}>
            <MenuItem test-id="LinkedIn" style={theme.sidebar} onClick={() => handleIconClick("https://www.linkedin.com/in/hardik-gaur-135891122/", "_blank")}>
               <LinkedIn
                  style={{ color: theme.icons.linkedin.color, fontSize: 35, marginRight: 20, }} />
               <span style={theme.text}>Linkedin</span>
            </MenuItem>
            <MenuItem test-id="Mail" style={theme.sidebar} onClick={() => handleIconClick("mailto:hardik3296@outlook.com", "_self")}>
               <Mail
                  style={{ color: theme.icons.mail.color, fontSize: 35, marginRight: 20, }}
               />
               <span style={theme.text}>Mail</span></MenuItem>
            <MenuItem test-id="GeeksForGeeks" style={theme.sidebar} onClick={() => handleIconClick("https://auth.geeksforgeeks.org/user/HGaur/articles", "_blank")} >
               <GeeksForGeeks
                  width={35}
                  height={35}
                  color={theme.icons.mail.color}
               />
               <span style={iconStyle}>Geeksforgeeks</span></MenuItem>
            <MenuItem test-id="Github" style={theme.sidebar} onClick={() => handleIconClick("https://github.com/Hardik3296", "_blank")}>
               <GitHub
                  style={{ color: theme.icons.github.color, fontSize: 35, marginRight: 20, }} />
               <span style={theme.text}>Github</span></MenuItem>
         </Menu>
      </>
   );
}

export default DrawerComponent;