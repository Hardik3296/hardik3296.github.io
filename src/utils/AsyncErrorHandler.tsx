import React from "react";

const useAsyncError = () => {
   const [_error, setError] = React.useState();
   return React.useCallback(
      e => {
         setError(() => {
            throw e;
         });
      },
      [setError],
   );
};

export default useAsyncError;