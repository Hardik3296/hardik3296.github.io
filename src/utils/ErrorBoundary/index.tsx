import { Component, ReactNode } from "react";
import styles from "./styles.module.scss";

interface MyProps {
   textcolor: string,
   children: ReactNode,
}

class ErrorBoundary extends Component<MyProps, {}> {

   state = {
      hasError: false
   }

   static getDerivedStateFromError() {
      return { hasError: true }
   }

   componentDidCatch(error: any, errorInfo: any) {
   }
   render() {
      if (this.state.hasError)
         return <div className={styles.errorDiv}>
            <p style={{ color: this.props.textcolor }}>Something went wrong while fetching the repositories!!</p>
         </div>
      else
         return this.props.children;
   }
}

export default ErrorBoundary;