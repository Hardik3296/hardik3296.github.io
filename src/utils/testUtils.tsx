import { ShallowWrapper } from "enzyme"

export const findAttributeByTestId = (component:ShallowWrapper, attr:string):ShallowWrapper=>{
   return component.find(`[test-id='${attr}']`);
}