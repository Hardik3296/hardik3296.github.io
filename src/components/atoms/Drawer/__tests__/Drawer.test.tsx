import {shallow, ShallowWrapper} from "enzyme";
import Drawer from "..";
import {findAttributeByTestId} from "../../../../utils/testUtils";

const renderComponent=()=>{
   return shallow(<Drawer />);
}

describe("Drawer Component",()=>{

   let wrapper:ShallowWrapper;

   beforeEach(()=>{
      wrapper=renderComponent();
   })

   it("Should have one linkedin icon",()=>{
      const component = findAttributeByTestId(wrapper,"LinkedIn");
      expect(component.length).toBe(1);
   });

   it("Should have one github icon",()=>{
      const component = wrapper.find(`[test-id="Github"]`);
      expect(component.length).toBe(1);
   });

   it("Should have one mail icon",()=>{
      const component = wrapper.find(`[test-id="Mail"]`);
      expect(component.length).toBe(1);
   });

   it("Should have one GeeksForGeeks icon",()=>{
      const component = wrapper.find(`[test-id="GeeksForGeeks"]`);
      expect(component.length).toBe(1);
   });

   it("renders correctly",()=>{
      expect(wrapper).toMatchSnapshot();
   });
});