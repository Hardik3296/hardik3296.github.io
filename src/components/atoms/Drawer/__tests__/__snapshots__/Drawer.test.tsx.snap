// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Component renders correctly 1`] = `ShallowWrapper {}`;
